csharp/src/Microsoft.ML.OnnxTraining/TrainingSession.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.ML.OnnxRuntime;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.OnnxTraining
+{
+    public class TrainingSession
+    {
+        TrainingParameters m_param = new TrainingParameters();
+
+        public TrainingSession()
+        {
+        }
+
+        public TrainingParameters Parameters
+        {
+            get { return m_param; }
+        }
+
+        public void Initialize(LogLevel logLevel = LogLevel.Warning)
+        {
+            IntPtr hEnv;
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateEnv(logLevel, @"CSharpOnnxRuntime.Training", out hEnv));
+            NativeApiStatus.VerifySuccess(NativeMethodsTraining.OrtInitializeTraining(hEnv, m_param.Handle));
```

Comment:
>OrtInitializeTraining [](start = 64, length = 21)

What happens to hEnv if OrtInitializeTraining returns an error and throws? Is it leaking?

Reply:
Agreed - using OrtEnv would should this.